lib/devices/gadgets/fitness_equipment.dart

```diff
@@ -392,62 +393,65 @@ class FitnessEquipment extends DeviceBase {
     if (_equipmentDiscovery || descriptor == null) return false;
 
     _equipmentDiscovery = true;
-    // Check manufacturer name
+
     if (manufacturerName == null) {
       final deviceInfo = BluetoothDeviceEx.filterService(services, deviceInformationUuid);
-      final nameCharacteristic =
-          BluetoothDeviceEx.filterCharacteristic(deviceInfo?.characteristics, manufacturerNameUuid);
-      if (nameCharacteristic == null) {
-        return false;
-      }
+      await _getManufacturerName(deviceInfo);
+    }
 
-      try {
-        final nameBytes = await nameCharacteristic.read();
-        manufacturerName = String.fromCharCodes(nameBytes);
-      } on PlatformException catch (e, stack) {
-        if (_logLevel > logLevelNone) {
-          Logging.logException(
-            _logLevel,
-            "FITNESS_EQUIPMENT",
-            "discover",
-            "Could not read name",
-            e,
-            stack,
-          );
-        }
+    _equipmentDiscovery = false;
+    return _checkManufacturerName();
+  }
 
-        if (kDebugMode) {
-          debugPrint("$e");
-          debugPrintStack(stackTrace: stack, label: "trace:");
-        }
+  bool _checkManufacturerName() {
+    if (_logLevel >= logLevelInfo) {
+      Logging.log(
+        _logLevel,
+        logLevelInfo,
+        "FITNESS_EQUIPMENT",
+        "_checkManufacturerName",
+        "ensuring that manufacturer name ($manufacturerName) contains manufacturer prefix ${descriptor!.manufacturerPrefix}",
+      );
+    }
+    return manufacturerName!.toLowerCase().contains(descriptor!.manufacturerPrefix.toLowerCase()) ||
```

Comment:
Can we make this robust against `manufacturerName` being `null`? In that case we'd return false.

Reply:
Done.